buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -706,6 +706,66 @@ public static void copy(AsciiString src, int srcIdx, ByteBuf dst, int length) {
     }
 
     /**
+     * Creates a retained duplicate of the specified {@link ByteBuf}.
+     */
+    public static ByteBuf rduplicate(ByteBuf buf) {
+        final int readerIndex = buf.readerIndex();
+        final int writerIndex = buf.writerIndex();
+
+        final AbstractByteBuf duplicate;
+        if (buf instanceof SlicedAbstractByteBuf) {
+            final SlicedAbstractByteBuf oldSlice = (SlicedAbstractByteBuf) buf;
+            duplicate = PooledSlicedByteBuf.newInstance(oldSlice.unwrap(), 0, buf.capacity(), oldSlice.adjustment());
```

Comment:
Well you are right, but I mean that it's supposed to be a way to get a pooled duplicate not a shortcut for duplicate().retain(). So by using rduplicate() you request a pooled slice if possible. In either way it's transparent for the API contract even if it would return a non-pooled one instead.

And the reason why it's not part of the allocator is simple - it's clearly not supposed to take care of slices and duplicates in current design.

But I think I can see your point...we could make sure that rduplicate() only uses pooled buffers if the parent is a pooled one and shortcuts to duplicate().retain() for non-pooled ones, if that's going to be useful in anything.